mlops-cdk-github-action/README.md

```diff
@@ -0,0 +1,163 @@
+
+## Solution Architecture
+
+![mlops project architecture](diagrams/github_action_mlops_architecture.jpg)
+
+
+## Repository Structure
+
+```
+.
+├── LICENSE.txt
+├── Makefile
+├── README.md
+├── app.py
+├── cdk.json
+├── diagrams
+├── .github                                   <--- contains the GitHub Action WorkFlow script
+│   ├── sm_template_register_service_catalog.yml  
+├── mlops_sm_project_template
+│   ├── README.md
+│   ├── __init__.py
+│   ├── cdk_helper_scripts
+│   ├── config
+│   │   └── constants.py                      <--- global configs to be used in CDK stacks
+│   ├── core_stage.py                     <--- entry to build the differnt stacks
+│   ├── service_catalog_stack.py              <--- stack for service catalog setup and template deployment
+│   └── templates
+│       ├── basic_project_stack.py                <--- stack for basic sagemaker project template setup - DEV Accounts provided in constants.py
+│       └── pipeline_constructs
+│           ├── build_pipeline_construct.py       <--- construct containing CI/CD pipeline linked to the build app
+│           └── deploy_pipeline_construct.py      <--- construct containing CI/CD pipeline linked to the deploy app
+├── requirements-dev.txt
+├── requirements.txt                          <--- cdk packages used in the stacks (must be installed)
+├── scripts                                   <--- shell scripts to automate part of the deployments
+│   ├── cdk-account-setup.sh
+│   └── install-prerequisites-brew.sh
+└── seed_code                                 <--- code samples to be used to setup the build and deploy repositories of the sagemaker project
+    ├── build_app
+    └── deploy_app
+```
+
+## Deployment Options
+
+There are two deployment options for the SageMaker Project Template to the Service Catalog in the target account:
+
+- **[CI/CD Deployment of Service Catalog Stack](#cicd-deployment-of-service-catalog-stack)** - deploy by using GitHub Action CICD pipeline by cloning the repository to your GitHub Repository
+
+- **[Manual Deployment of Service Catalog Stack](#manual-deployment-of-service-catalog-stack)** - deploy directly to the targeted accounts using CDK commands from your local development setup
+
+
+
+### CI/CD Deployment of Service Catalog Stack
+Register the SageMaker Project Template through GitHub Action Pipeline CI/CD ( Preferred )
+
+Follow below steps:
+
+1. Create AWS GitHub OpenId connect IAM role, follow the below process : https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services
+
+2. Create following GitHub secrets : AWS_ACCOUNT_OPENID_IAM_ROLE, AWS_REGION, AWS_ACCOUNT
+
+3. GitHub Action deployment workflow CI/CD pipeline (.github/workflows/sm_template_register_service_catalog.yml) will now handle all deployments.
+
+
+
+
+### Manual Deployment of Service Catalog Stack
+Register the SageMaker Project Template from local development.
+
+#### Pre-requisites
```

Comment:
As discussed please add prerequisites for:
For deploying Service Catalog: create SSM Parameters /mlops/sagemaker/execution with value used in SageMaker Studio domain
For creating SageMaker Project/Service Catalog product, provide short on how to create github token with which actions are required

Reply:
Requirements: create Identity provider for Github to sts if not already there in account, following: https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services#adding-the-identity-provider-to-aws

Reply:
Could you also add high level instructions for how to enable multi account deployment?

Reply:
Updated the instructions on README file.